src/drivers/net/ena.c

```diff
@@ -206,23 +207,39 @@ static int ena_create_admin ( struct ena_nic *ena ) {
 	}
 	memset ( ena->aq.req, 0, aq_len );
 
+	/* Allocate AENQ queue */
+	ena->aenq.events = malloc_phys ( aenq_len, aenq_len );
+	if ( ! ena->aenq.events ) {
+		rc = -ENOMEM;
+		goto err_alloc_aenq;
+	}
+	memset ( ena->aenq.events, 0, aenq_len );
+
 	/* Program queue addresses and capabilities */
 	ena_set_base ( ena, ENA_ACQ_BASE, ena->acq.rsp );
 	ena_set_caps ( ena, ENA_ACQ_CAPS, ENA_ACQ_COUNT,
 		       sizeof ( ena->acq.rsp[0] ) );
 	ena_set_base ( ena, ENA_AQ_BASE, ena->aq.req );
 	ena_set_caps ( ena, ENA_AQ_CAPS, ENA_AQ_COUNT,
 		       sizeof ( ena->aq.req[0] ) );
+	ena_set_base ( ena, ENA_AENQ_BASE, ena->aenq.events );
+	ena_set_caps ( ena, ENA_AENQ_CAPS, ENA_AENQ_COUNT,
+		       sizeof ( ena->aenq.events[0] ) );
 
-	DBGC ( ena, "ENA %p AQ [%08lx,%08lx) ACQ [%08lx,%08lx)\n",
+	DBGC ( ena, "ENA %p AQ [%08lx,%08lx) ACQ [%08lx,%08lx) AENQ [%08lx,%08lx)\n",
 	       ena, virt_to_phys ( ena->aq.req ),
 	       ( virt_to_phys ( ena->aq.req ) + aq_len ),
 	       virt_to_phys ( ena->acq.rsp ),
-	       ( virt_to_phys ( ena->acq.rsp ) + acq_len ) );
+	       ( virt_to_phys ( ena->acq.rsp ) + acq_len ),
+	       virt_to_phys ( ena->aenq.events ),
+	       ( virt_to_phys ( ena->aenq.events ) + aenq_len ));
 	return 0;
 
 	ena_clear_caps ( ena, ENA_AQ_CAPS );
 	ena_clear_caps ( ena, ENA_ACQ_CAPS );
+	ena_clear_caps ( ena, ENA_AENQ_CAPS );
```

Comment:
The corresponding call to `ena_clear_caps()` in `ena_destroy_admin()` is missing.